src/shared/formatters.ts

```diff
@@ -112,6 +112,41 @@ export function humanizeDurationFromSeconds(duration: number): string {
   return humanizeDuration(duration * 1000);
 }
 
+export function heikinAshiDataset(columns: string[], data: Array<number[]>) {
+  const openIdx = columns.indexOf('open');
+  const closeIdx = columns.indexOf('close');
+  const highIdx = columns.indexOf('high');
+  const lowIdx = columns.indexOf('low');
+
+  // Prevent mutation of original data
+  const dataCopy = data.map((original) => original.slice());
+
+  return dataCopy.map((candle, idx, candles) => {
```

Comment:
Just one question - how will this scale to more data?

in webserver mode (`freqtrade webserver ...`) - you can get 1 year of 5m data into the chart - which currently renders quite ok from a performance perspective.

will we expect a "hang" if we click that button in that case? (if yes - we might need to disable the button for backtesting mode - or add a warning in that mode).

Reply:
It seems to perform just fine. I straight up imported the json file of 1y-of-5m to the frequi and loaded it directly to the chart.

![heikin-ashi-perf](https://user-images.githubusercontent.com/11251134/157861834-1c527cea-8fa1-4bfa-9cc1-f23f0f6af667.gif)

Reply:
Nice - thanks for checking! :+1: